File: Scala/test/appendix/MinimizingBlockingTests.scala
package appendix

import org.scalatest.funsuite.AnyFunSuite
import tinyscalautils.threads.shutdownAndWait

class MinimizingBlockingTests extends AnyFunSuite:
   import MinimizingBlockingJ.*

   test("termination/deadlock") {
      for n <- 1 to 10; m <- 1 to 10 do assert(demo(n, m).shutdownAndWait(0.5) == m <= n)
   }

File: Kotlin/test/appendix/MinimizingBlockingTests.kt
package appendix

import appendix.MinimizingBlocking.demo
import kotlinx.coroutines.runBlocking
import kotlin.test.Test

class MinimizingBlockingTests {
  @Test
  fun testCountDownLatch() {
    for (n in 1..10)
      for (m in 1..10)
        runBlocking {
          demo(n, m).shutdown()
        }
  }
}
